rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/CourseOfStudyService.java

```diff
@@ -37,6 +37,16 @@
 
     public GetCourseOfStudiesResponse getAllResponse() {
         Set<CourseOfStudy> courseOfStudies = new HashSet<>(this.courseOfStudyRepository.findAll());
+        Set<CourseOfStudy> predecessors = new HashSet<>();
+
+        for (CourseOfStudy courseOfStudy : courseOfStudies) {
+            CourseOfStudy predecessor = courseOfStudy.getPredecessor();
+            if (predecessor != null) {
+                predecessors.add(predecessor);
+            }
```

Comment:
Shouldn't we remove all predecessors iteratively?

Reply:
Ah, nevermind, I get it now

Reply:
There may be the same predecessor for multiple courses, what then? When we are adding it to the Set it actually isn't added at all, with delete statement we would need probably add another "if predecessor in courses", so I don't think it is good idea, which path do you prefer?